src/models/activityReportGoal.js

```diff
@@ -1,6 +1,7 @@
 const { Model } = require('sequelize');
 const { CLOSE_SUSPEND_REASONS } = require('../constants');
 const { formatDate } = require('../lib/modelHelpers');
+const { afterDestroy } = require('./hooks/activityReportGoal');
```

Comment:
Looks like we are adding 'afterDestroy' to a bunch of models. Why was the decision made to update the various fields from this function as opposed to using an existing one like 'afterUpdate'?

Reply:
when a goal is removed from a report, we check to see if its still on any other report, then populate if a change needs to occur. The same goes for any other entity hanging off of a report.